### PR TITLE
Rolldown Hierarchy

### PR DESCRIPTION
**Pull request by amritanarasimhan**
shift+click for to see all elements in hierarchy. Also fixed issues with a previous PR about the scene tree not sticking

### CODE REVIEW

File: packages/tools/guiEditor/src/components/sceneExplorer/treeItemSelectableComponent.tsx [hunk 1]
```diff
@@ -138,7 +147,7 @@ export class TreeItemSelectableComponent extends React.Component<ITreeItemSelect
         const marginStyle = {
             paddingLeft: 10 * (this.props.offset + 0.5) - 20 + "px",
         };
-        const entity = this.props.entity;
+        let entity = this.props.entity;
```
Comment (RaananW) [inline packages/tools/guiEditor/src/components/sceneExplorer/treeItemSelectableComponent.tsx:150]: just wondering - what is the reason for this change?

**Review by RaananW — COMMENTED** [review]
LGTM! just two questions. I'll also pull the branch and try it first before approving :-)

File: packages/tools/guiEditor/src/components/sceneExplorer/treeItemSelectableComponent.tsx (hunk 1 above)
Comment (amritanarasimhan) [reply packages/tools/guiEditor/src/components/sceneExplorer/treeItemSelectableComponent.tsx:150]: In the section of code starting at line 230 I change the value of entity, which is why I changed it from const to let. Would it be better practice to create a separate entity variable that I change?

File: packages/tools/guiEditor/src/diagram/workbench.tsx [hunk 2]
```diff
@@ -286,6 +286,8 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
                 for (const control of parsed.controls) {
                     newSelection.push(this.appendBlock(Control.Parse(control, this.props.globalState.guiTexture)));
                 }
+                newSelection[0].leftInPixels = newSelection[0].leftInPixels + 10;
+                newSelection[0].topInPixels = newSelection[0].topInPixels + 10;
```
[the remark this answers is not in the document]
Comment (amritanarasimhan) [reply packages/tools/guiEditor/src/diagram/workbench.tsx:290]: Oh no! That file shouldn't have been there at all, as those changes are for another issue. The only file I changed for this issue was treeItemSelectableComponent,

[the remark this answers is not in the document]
Comment (amritanarasimhan) [reply packages/tools/guiEditor/src/diagram/workbench.tsx:290]: Would it be alright if I deleted this file from the PR? I checked in the branch for that issue, and these changes are properly saved there.

[the remark this answers is not in the document]
Comment (RaananW) [reply packages/tools/guiEditor/src/diagram/workbench.tsx:290]: Let's discuss this offline later :-)

File: packages/tools/guiEditor/src/components/sceneExplorer/treeItemSelectableComponent.tsx (hunk 1 above)
Comment (RaananW) [reply packages/tools/guiEditor/src/components/sceneExplorer/treeItemSelectableComponent.tsx:151]: no no, this is fine. I just didn't see the line change that re-sets the variable

**Review by RaananW — APPROVED** [review]
